doc/operator-capabilities.md

```diff
@@ -0,0 +1,193 @@
+
+Each capability level is associated with a certain set of management features the Operator offers around the managed workload. Operator that do not manage a workload and/or are delegating to off-clusters orchestration services would remain at Level 1. Capability levels are accumulating, i.e. Level 3 capabilities require all capabilities desired from Level 1 and 2.
+
+
+## Terminology
+
+**Operator** - the custom controller installed on a Kubernetes cluster
+
+**Operand** - the managed workload provided by the Operator as a service
+
+**Custom Resource (CR)** - an instance of the `CustomResourceDefinition` the Operator ships that represents the Operand or an Operation on an Operand
+
+**Managed resources** - the Kubernetes objects or off-cluster services the Operator uses to constitute an Operand
+
+---
+
+## Level 1 - Basic Install
+
+The Operator offers the following basic features:
+
+**Installation of the workload**
+
+- Operator deploys an Operand or configures off-cluster resources
+- Operator waits for managed resources to reach a healthy state
+- Operator conveys readiness of application or managed resources to the user leveraging the `status` block of the Custom Resource
+
+**Example:** an Operator deploys a database by creating `Deployment`, `ServiceAccount`, `RoleBinding`, `ConfigMap`, `PersistentVolumeClaim` and `Secret` object, initializes an empty database schema and signals readiness of the database to accept queries.
+
+**Configuration of the workload**
+
+- Operator provides configuration via the `spec` section of the Custom Resource
+- Operator reconciles configuration and updates to it with the status of the managed resources
+
+**Example:** an Operator, managing a database, offers increasing the capacity of the database by resizing the underlying `PersistentVolumeClaim` based on changes the databases Custom Resource instance.
+
+**Guiding questions to determine Operator reaching Level 1**
+
+1. What installation configuration can be set in the CR?
+
+2. What additional CR installation configuration could still be added?
+
+3. Can you set operand configuration in the CR? If so, what configuration is supported for each operand?
+
+4. Does the managed application / workload get updated in a non-disruptive fashion when the configuration of the CR is changed?
+
+5. Does the status of the CR reflect that configuration changes are currently applied?
+
+6. What additional operand configuration could still be added?
+
+7. Do all of the instantiated CRs include a status block? If so, does it provide enough insight to the user about the application state?
+
+8. Do all of your CRs have documentation listing valid values and mandatory fields?
+
+---
+
+## Level 2 - Seamless Upgrades
+
+The Operator offers the following features related to upgrades:
+
+**Upgrade of the managed workload**
+
+- Operand can be upgraded in the process of upgrading the Operator, or
+- Operand can be upgraded as part of changing the CR
```

Comment:
is not better use API instead of CRD. 
Note that changes in the CR's might has no relation with the upgrade of the operator and or its APIs ( CRD's )

Reply:
I am not sure what the proposed action item here is. This section does not mention CRDs, nor does it create a relation between upgrade of the Operator and its CRDs. Is this something that people would infer?

Reply:
When we change/update a CR we are not doing an upgrade of the operator.

Reply:
Yes, and the cited section talks about the upgrade of the `Operand` as part of changing/updating a CR.